src/zm_logger.cpp

```diff
@@ -529,15 +529,16 @@ void Logger::logPrint(bool hex, const char *filepath, int line, int level, const
   if (level <= mDatabaseLevel) {
     if (zmDbConnected) {
       int syslogSize = syslogEnd-syslogStart;
-      char escapedString[(syslogSize*2)+1];
-      mysql_real_escape_string(&dbconn, escapedString, syslogStart, syslogSize);
+      std::string escapedString;
+      escapedString.reserve((syslogSize * 2) + 1);
+      mysql_real_escape_string(&dbconn, &escapedString[0], syslogStart, syslogSize);
```

Comment:
Isn't this considered bad form writing to the array address of the std::string?

Reply:
Since C++11 this is actually supported since the standard guarantees a continuous internal buffer. C++17 even added a non-const overload for `.data()` which makes this even nice to read.

I added one small thing (would have worked without it in our use-case): Shrink the string to `strlen` of the escpaed one so that `size()` works as well.